### ISSUE TITLE
Using --branch and git-branch in autoversion crashes

### ISSUE THREAD

**Issue by Linuus** [boldpoker/edeliver#103]
Hi!

When I have this in my config:
```sh
AUTO_VERSION=commit-count+git-revision+branch-unless-master
```
and I run `mix edeliver build release` it works fine and it builds from the latest commit in the master branch.

But if I check out a new branch, add a new commit here and then run `mix edeliver build release --branch=something` it crashes with this error:

```sh
==> ERROR: "Failed to build release. Please fix any errors and try again."

A remote command failed on:

  linus@localhost

Output of the command is shown above and the command executed
on that host is printed below for debugging purposes:

FAILED with exit status 1:

    [ -f ~/.profile ] && source ~/.profile
    set -e
    cd /tmp/erlang/chat/builds
    if [ "mix" = "rebar" ]; then
      echo "using rebar to generate release"
      ./rebar   -f generate
    elif [ "mix" = "relx" ]; then
      echo "using relx to generate release"
      ./relx release
    elif [ "mix" = "mix" ]; then
      echo "using mix to generate release"
      MIX_ENV="prod" LINK_SYS_CONFIG="" LINK_VM_ARGS="" APP="chat" AUTO_VERSION="commit-count+git-revision+branch-unless-master" BRANCH="something" SKIP_RELUP_MODIFICATIONS="" RELUP_MODIFICATION_MODULE="" mix do release.version  ,  release  --no-confirm-missing --verbosity=verbose
    fi
```

If I remove `branch-unless-master` from the `AUTOVERSION` variable it works, but I think it's nice to see from what branch a release has been built.

Any ideas why this might happen?

Also, when I have branch names like `feature/some-feature` it complains about `** (Mix) Expected :version to be a SemVer version`. I guess that's due to the `/`? Would it be possible to work around this somehow (chaning the branch names style is probably not possible...)?

**Comment by bharendt** [comment]
Your are right, invalid characters in the branch name are [not removed](https://github.com/boldpoker/edeliver/blob/4de6d7dcb2738dd311f4b7c31033b88eacd07db0/lib/mix/tasks/release_version.ex#L348) correctly any more. This was introduced in 4de6d7dcb2738dd311f4b7c31033b88eacd07db0 which was necessary, because a commit can exist in serveral branches (see #95 and #74). Thanks for reporting this.

**Comment by bharendt** [comment]
This fix is included in edeliver version [edeliver version 1.2.10](https://hex.pm/packages/edeliver/1.2.10)
